lib/dom.js

```diff
@@ -606,14 +606,27 @@ Document.prototype = {
 		return rtv;
 	},
 	
-	getElementsByClassName: function(className) {
-		var pattern = new RegExp("(^|\\s)" + className + "(\\s|$)");
+	getElementsByClassName: function(classNames) {
+		var spaceSeparatedTokens = function(string) {
+			// 5 space characters, as defined in html spec
+			return string.split(/[ \t\n\f\r]+/).filter(function(name) {
+				return name !== ''
+			})
+		}
+		var namesTarget = spaceSeparatedTokens(classNames)
 		return new LiveNodeList(this, function(base) {
 			var ls = [];
 			_visitNode(base.documentElement, function(node) {
 				if(node !== base && node.nodeType == ELEMENT_NODE) {
-					if(pattern.test(node.getAttribute('class'))) {
-						ls.push(node);
+					var classActual = node.getAttribute('class')
+					if (classActual && namesTarget.length > 0) { // don't match '' for ''
+						var namesActual = spaceSeparatedTokens(classActual)
+						var matches = namesTarget.every(function(name) {
+							return namesActual.indexOf(name) !== -1
+						})
```

Comment:
I guess that "the most common use case" would be to match a single class name (or even all class names of an element and with some luck in the right order), which could still succeed by just comparing them:
```
// before splitting and iterating just compare them for the most common use case(s)
if (classNames === classActual) {
	ls.push(node);
} else {
	// split and compare as you do above
	if(matches) (
		ls.push(node);
	}
}
```
it's of course an optimization, but I would guess that's it could be worth it performance wise.